lib/twirp/rails/rescuable.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Twirp
+  module Rails
+    module Rescuable
+      refine ::ActiveSupport::Rescuable::ClassMethods do
+        # A slightly altered version of ActiveSupport::Rescuable#rescue_with_handler
+        # that returns the result rather than the handled exception
+        def rescue_with_handler_and_return(exception, object: self, visited_exceptions: [])
+          visited_exceptions << exception
+
+          if (handler = handler_for_rescue(exception, object: object))
+            handler.call exception
+          elsif exception
+            if visited_exceptions.include?(exception.cause)
+              nil
+            else
+              rescue_with_handler(exception.cause, object: object, visited_exceptions: visited_exceptions)
```

Comment:
```suggestion
          if (handler = handler_for_rescue(exception, object: object))
            handler.call exception
          elsif exception
            if visited_exceptions.include?(exception.cause)
              nil
            else
              rescue_with_handler(exception.cause, object: object, visited_exceptions: visited_exceptions << exception)
```

Reply:
That's better, but the original method was copied out of ActiveSupport so I'm tempted to leave it as close as possible for easier comparison in the future.